runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/source/unbounded/FlinkUnboundedSourceReader.java

```diff
@@ -103,22 +104,23 @@ public void start() {
     createPendingBytesGauge(context);
     Long watermarkInterval =
         pipelineOptions.as(FlinkPipelineOptions.class).getAutoWatermarkInterval();
-    if (watermarkInterval != null) {
-      scheduleTaskAtFixedRate(
-          () -> {
-            // Set the watermark emission flag first.
-            shouldEmitWatermark = true;
-            // Wake up the main thread if necessary.
-            CompletableFuture<Void> f = dataAvailableFutureRef.get();
-            if (f != DUMMY_FUTURE) {
-              f.complete(null);
-            }
-          },
-          watermarkInterval,
-          watermarkInterval);
-    } else {
-      LOG.warn("AutoWatermarkInterval is not set, watermarks won't be emitted.");
+    if (watermarkInterval == null) {
+      watermarkInterval = DEFAULT_WATERMARK_EMIT_INTERVAL_MS;
```

Comment:
Can we derive this from `getMaxBundleTimeMillis()` from `FlinkPipelineOptions`? It might be maxBundleTimeMillis / 5 (to get the 200 ms default) or so.

Reply:
Thanks for the feedback. I think I've fixed this now.

Reply:
Hi @je-ik could you review again?